ext/span.c

```diff
@@ -796,3 +823,75 @@ zend_string *ddtrace_trace_id_as_hex_string(ddtrace_trace_id id) {
     snprintf(ZSTR_VAL(str), 33, "%016" PRIx64 "%016" PRIx64, id.high, id.low);
     return str;
 }
+
+
+ddtrace_root_span_data *ddtrace_open_inferred_span(zend_array *headers) {
+    ddtrace_read_header *read_header = headers ? ddtrace_read_array_header : ddtrace_read_zai_header;
+    ddtrace_inferred_proxy_result result = ddtrace_read_inferred_proxy_headers(read_header, headers);
+
+    if (!result.system || !result.start_time_ms) {
+        if (result.system) zend_string_release(result.system);
+        if (result.start_time_ms) zend_string_release(result.start_time_ms);
+        return NULL;
+    }
+
+    const ddtrace_proxy_info *proxy_info = ddtrace_get_proxy_info(result.system);
+    if (!proxy_info) {
+        zend_string_release(result.system);
+        zend_string_release(result.start_time_ms);
+        return NULL;
+    }
+
+    ddtrace_span_data *span = ddtrace_push_inferred_root_span();
+
+    zval_ptr_dtor(&span->property_name);
+    ZVAL_STR(&span->property_name, zend_string_init(proxy_info->span_name, strlen(proxy_info->span_name), 0));
+
+    zval_ptr_dtor(&span->property_resource);
+    if (result.http_method && result.path) {
+        ZVAL_STR(&span->property_resource, strpprintf(0, "%s %s", ZSTR_VAL(result.http_method), ZSTR_VAL(result.path)));
+    }
+
+    span->start = (zend_long)zend_strtod(ZSTR_VAL(result.start_time_ms), NULL) * 1000000;
```

Comment:
This looks fishy. Either you should directly parse it as a long, or you need parenthesis (if the number is actually a decimal number). The (zend_long) cast is evaluated before the multiplication.
If the goal is string to zend_long, use `ZEND_ATOL()`.